main/networking/Ansible.js

```diff
@@ -6,86 +6,76 @@ import _ from 'lodash';
 
 import RendererBridge from '../RendererBridge';
 import { updateConsole } from '../../renderer/actions/ConsoleActions';
-import {
-  ansibleDisconnect,
-  infoPerMessage,
-  updateCodeStatus,
-} from '../../renderer/actions/InfoActions';
+import { ansibleDisconnect } from '../../renderer/actions/InfoActions';
 import { updatePeripherals } from '../../renderer/actions/PeripheralActions';
-import { robotState, Logger, defaults } from '../../renderer/utils/utils';
-import FCObject from './FieldControl';
-
-const DawnData = (new protobuf.Root()).loadSync(`${__dirname}/ansible.proto`, { keepCase: true }).lookupType('DawnData');
-const { StudentCodeStatus } = DawnData;
-
-const RuntimeData = (new protobuf.Root()).loadSync(`${__dirname}/runtime.proto`, { keepCase: true }).lookupType('RuntimeData');
-const Notification = (new protobuf.Root()).loadSync(`${__dirname}/notification.proto`, { keepCase: true }).lookupType('Notification');
+import { Logger, defaults } from '../../renderer/utils/utils';
+
+/**
+ * UDP Recv (Runtime -> Dawn)
+ * Device Data Array (sensors), device.proto
+ */
+const RecvDeviceProto = (new protobuf.Root()).loadSync('protos/device.proto', { keepCase: true }).lookupType('DevData');
+
+/**
+ * UDP Send (Dawn -> Runtime)
+ * Gamepad Data, gamepad.proto
+ */
+const SendGamepadProto = (new protobuf.Root()).loadSync('protos/gamepad.proto', { keepCase: true }).lookupType('GpState');
+
+/**
+ * TCP Recv (Runtime -> Dawn)
+ * Challenge Data (output values), text.proto
+ * Log Data (console), text.proto
+ */
+const RecvChallengeProto = (new protobuf.Root()).loadSync('protos/text.proto', { keepCase: true }).lookupType('Text');
+const { RecvLogProto } = RecvChallengeProto;
+
+/**
+ * TCP Send (Dawn -> Runtime)
+ * Run Mode Data (TELEOP, ESTOP), run_mode.proto
+ * Device Data Array (preference filter), device.proto
+ * Challenge Data (input values), text.proto
+ * Position Data, start_pos.proto
+ */
+const SendModeProto = (new protobuf.Root()).loadSync('protos/run_mode.proto', { keepCase: true }).lookupType('RunMode');
+const { SendDeviceProto } = RecvDeviceProto;
+const { SendChallengeProto } = RecvChallengeProto;
+const SendPosProto = (new protobuf.Root()).loadSync('protos/start_pos.proto', { keepCase: true }).lookupType('StartPos');
 
 const LISTEN_PORT = 1235;
 const SEND_PORT = 1236;
 const TCP_PORT = 1234;
 
-function buildProto(data) {
-  let status = null;
-  switch (data.studentCodeStatus) {
-    case robotState.TELEOP:
-      status = StudentCodeStatus.TELEOP;
-      break;
-    case robotState.AUTONOMOUS:
-      status = StudentCodeStatus.AUTONOMOUS;
-      break;
-    case robotState.ESTOP:
-      status = StudentCodeStatus.ESTOP;
-      break;
-    default:
-      status = StudentCodeStatus.IDLE;
-  }
-
+function buildGamepadProto(data) {
   const gamepads = _.map(_.toArray(data.gamepads), (gamepad) => {
```

Comment:
This is what I was referring to a while ago in a DM--is it possible for there to be multiple gamepad**s** to be connected to Dawn simultaneously?

Reply:
From what I've seen at the events, it was always 4 stations each with 1 gamepad and each running Dawn, so I've never seen multiple gamepads, but we can definitely experiment to see if this is possible.

Reply:
No, it would be good if it was just one gamepad per computer--Runtime can't handle more than one gamepad connected to Dawn as of right now.... I don't think it's necessary to allow for more than one gamepad to be connected at once

Reply:
This probably should be a feature in Runtime @benliao1 . A future enhancement we can add